website/project/model.py

```diff
@@ -2611,12 +2611,15 @@ def add_unregistered_contributor(self, fullname, email, auth,
         self.save()
         return contributor
 
-    def set_privacy(self, permissions, auth=None, log=True, save=True):
-        """Set the permissions for this node.
+    def set_privacy(self, permissions, auth=None, log=True, save=True, skip_mail=False):
+        """Set the permissions for this node. Also, based on skip_mail, queues an email to user about abilities of
+            public projects.
 
         :param permissions: A string, either 'public' or 'private'
         :param auth: All the auth information including user, API key.
         :param bool log: Whether to add a NodeLog for the privacy change.
+        :param bool skip_mail: Whether this should trigger a new_public_project email, used to prevent conferences from
```

Comment:
This shouldn't be necessary; the raison d'etre for signals is that senders don't have to know about its listeners. If you need to bypass some listener behavior, then you should either (1) add a conditional to your listener or (2) disconnect from the listener when necessary. Option 1 is preferred.